src/hb-ot-layout-gsubgpos.hh

```diff
@@ -3678,55 +3678,60 @@ struct GSUBGPOS
                                 const hb_set_t *feature_indices,
                                 hb_map_t *duplicate_feature_map /* OUT */) const
   {
+    hb_set_t unique_features;
+    hb_tag_t prev_t = get_feature_tag (feature_indices->get_min ());
     //find out duplicate features after subset
-    unsigned prev = 0xFFFFu;
     for (unsigned i : feature_indices->iter ())
     {
-      if (prev == 0xFFFFu)
-      {
-        duplicate_feature_map->set (i, i);
-        prev = i;
-        continue;
-      }
-
       hb_tag_t t = get_feature_tag (i);
-      hb_tag_t prev_t = get_feature_tag (prev);
       if (t != prev_t)
       {
+        prev_t = t;
+        unique_features.clear ();
+        unique_features.add (i);
         duplicate_feature_map->set (i, i);
-        prev = i;
         continue;
       }
 
-      const Feature& f = get_feature (i);
-      const Feature& prev_f = get_feature (prev);
-
-      auto f_iter =
-      + hb_iter (f.lookupIndex)
-      | hb_filter (lookup_indices)
-      ;
-
-      auto prev_iter =
-      + hb_iter (prev_f.lookupIndex)
-      | hb_filter (lookup_indices)
-      ;
+      bool found = false;
 
-      if (f_iter.len () != prev_iter.len ())
+      for (unsigned other_f_index : unique_features.iter ())
       {
-        duplicate_feature_map->set (i, i);
-        prev = i;
-        continue;
-      }
-
-      bool is_equal = true;
-      for (auto _ : + hb_zip (f_iter, prev_iter))
-        if (_.first != _.second) { is_equal = false; break; }
+        const Feature& f = get_feature (i);
+        const Feature& other_f = get_feature (other_f_index);
+
+        auto f_iter =
+        + hb_iter (f.lookupIndex)
+        | hb_filter (lookup_indices)
+        ;
+  
+        auto other_f_iter =
+        + hb_iter (other_f.lookupIndex)
+        | hb_filter (lookup_indices)
+        ;
+  
+        if (f_iter.len () != other_f_iter.len ())
```

Comment:
This can be removed as

if (is_equal == false || f_iter || other_f_iter) continue; 

below is checking for equal lengths

Reply:
Good catch! I forgot to remove this line. :)